pkg/install/artifact/artifact.go

```diff
@@ -1,37 +1,408 @@
 package artifact
 
 import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
+	kustomizev1 "github.com/fluxcd/kustomize-controller/api/v1beta1"
+	"github.com/fluxcd/pkg/apis/meta"
+	"github.com/fluxcd/pkg/runtime/dependency"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	"github.com/otiai10/copy"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	kjson "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"knative.dev/pkg/apis"
 	"sigs.k8s.io/kustomize/api/types"
 )
 
-// Object is an artifact object which adds path to a runtime object.
-// path adds location detail to help generate this object.
-type Object struct {
-	runtime.Object
-	// Path, if set, will be used as folder location for this Object.
-	Path string
-	// Name, if given, will overwrite what this object's filename representation should be.
-	Name string
-}
-
-// Kustomize contains resources which direct flux to what to install.
-type Kustomize struct {
-	// LocalResourceLimiter resource which limits the number of objects scanned by flux.
-	LocalResourceLimiter *types.Kustomization
-	// ObjectWrapper is the top most Kubernetes object wrapper for all resources.
-	ObjectWrapper *types.Kustomization
-}
-
-// Artifact contains the name and objects belonging to a profile artifact
-type Artifact struct {
-	// Generated Kubernetes resources for this artifact.
-	Objects      []Object
-	Kustomize    Kustomize
-	Name         string
-	RepoURL      string
-	PathsToCopy  []string
-	SparseFolder string
-	Branch       string
-	SubFolder    string
+const (
+	defaultValuesKey           = "default-values.yaml"
+	helmChartLocation          = "helm-chart"
+	kustomizeWrapperObjectName = "kustomize-flux.yaml"
+)
+
+// ArtifactWriter can build an artifacts from an installation and a profile artifact.
+//go:generate counterfeiter -o fakes/fake_builder.go . ArtifactWriter
+type ArtifactWriter interface {
+	Write(installation profilesv1.ProfileInstallation, artifacts []ArtifactWrapper) error
+}
+
+var _ ArtifactWriter = &Writer{}
+
+type ArtifactWrapper struct {
```

Comment:
Missing comment. :) And there are some other Exported methods and structus which are missing comments. I won't tag them all. :D